### ISSUE TITLE
[Feature Request] allow changing the position of the image while using "resize and fill" resize mode

### ISSUE THREAD

**Issue by Hugs288** [Mikubill/sd-webui-controlnet#1920]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits of both this extension and the webui

### What happened?

Hello.
I was trying to get a full body version of this image in txt2img with controlnet inpaint by expanding it by 512 pixels downward. it is 512x512.
![01428-1486356993](https://github.com/Mikubill/sd-webui-controlnet/assets/131965855/cb415650-3146-47ea-8515-4802fe9b2c71)
as for my controlnet settings, i am just using the "inpaint" preset with the inpaint only+lama Preprocessor + pixel perfect + resize and fill resize mode, which are pretty standard settings for controlnet inpainting.
here are two non cherry picked results (512 -> 1024 height)
![00284-3490893545](https://github.com/Mikubill/sd-webui-controlnet/assets/131965855/cc5bd9f4-141c-4f2d-a671-9138e914a9a4)
![00278-293258150](https://github.com/Mikubill/sd-webui-controlnet/assets/131965855/1c77f5e5-fd48-4acb-94f8-0d3e1b99cc91)
the results are pretty good, but for some reason sd always insists on putting random nonesense on the top even though it should ideally be mostly a white background
another issue is that I want to expand the image by 512 pixels downward, not 256 pixels upward and downward.
both of these issues would be resolved if it were possible to change the position of the image while using the "resize and fill" resize mode.
here are some images made in paint to get my point across
Default resize and fill (currently the only option)
![default](https://github.com/Mikubill/sd-webui-controlnet/assets/131965855/8804468b-189c-44c9-8914-c048e4cde1d3)
resize and fill "top"
![top](https://github.com/Mikubill/sd-webui-controlnet/assets/131965855/466f7cad-b094-4438-a5ad-00da9e592c3e)
resize and fill "bottom"
![bottom](https://github.com/Mikubill/sd-webui-controlnet/assets/131965855/cdf6d127-12b1-40c2-bcba-fb30689355a4)
i think you can imagine the layouts for right, left, bottom right, etc.
ofcourse ideally we should be able to place the base image anywhere but these would still be very useful and (i think) fairly easy to implement.
i can achieve the images that I actually want by generating with 2048 height and then cropping out half of the image in with photo editing software, but that is unnecessarily slow, inconvenient, and produces weird artifacts.
![01431-1848616713](https://github.com/Mikubill/sd-webui-controlnet/assets/131965855/929ca119-c3d6-469f-9368-ce6755e79421)
![01430-2143732850](https://github.com/Mikubill/sd-webui-controlnet/assets/131965855/51e5ef42-d386-485e-874d-c18f38fb9b10)
(sd wouldnt do a white background no matter how hard i tried)
i would highly appreciate it.

### Steps to reproduce the problem

-

### What should have happened?

-

### Commit where the problem happens

-

### What browsers do you use to access the UI ?

Google Chrome

### Command Line Arguments

```Shell
-
```

### List of enabled extensions

-

### Console logs

```Shell
-
```

**Comment by huchenlei** [comment]
I don't think this is an issue. You can essentially resize the canvas in an image editor, mask the area you want to generative fill.

I think the more general thing you are asking it the ability to resize the canvas, like in an image editor. You can try the photopea extension in A1111 which allows you easily send images between photopea and A1111: https://github.com/yankooliveira/sd-webui-photopea-embed

Or just use A1111 in an image editor:
https://github.com/huchenlei/stable-diffusion-ps-pea
![](https://user-images.githubusercontent.com/20929282/256953922-4ae79838-ddeb-452a-a052-2b414381e709.png)
![](https://user-images.githubusercontent.com/20929282/256953923-f8406fcd-c7a6-4e81-9aeb-290e21755123.png)

**Comment by huchenlei** [comment]
Current workflow would be:

- In photopea, use the crop tool to expand the canvas the way you want.
- Send the image to A1111 img2img inpaint
- Mask the white area you just expanded in photopea
- Enable lama inpaint in ControlNet and click generate

Also, if you really want to specify which region to outpaint when selecting resize mode `resize and fill`, please submit an issue in A1111 repo, as the ControlNet output implementation depends on A1111 implementation.

**Comment by Hugs288** [comment]
all right, thank you for workflow
